File: 1929.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine(), " ");
		
		int m = Integer.parseInt(st.nextToken());
		int n = Integer.parseInt(st.nextToken());
		
		printPrimeNumbersBetweenMtoN(m, n);
	}
	
	
	private static void printPrimeNumbersBetweenMtoN(int m, int n) {
		boolean[] check = new boolean[n+1];
		
		check[0] = check[1] = true;
		
		for(int i=2; i<Math.sqrt(n); i++) {
			for(int j=i*i; j<=n; j+=i) {
				check[j] = true;
			}
		}

		StringBuilder sb = new StringBuilder();
		for(int i=m; i<=n; i++) {
			if(!check[i])
				sb.append(i).append("\n");
		}
		System.out.println(sb);
	}
}

File: 1929.c
#include<stdio.h>
#include<math.h>
int main(void){
	long long n;
	long long m;
	long long j;
	int w = 0, y = 0;
	scanf("%d", &m);
	scanf("%d", &n);
	for (long long i = m; i <= n; i++){
		if (i == 2){
			printf("%d\n", i);
		}
		else
		if (i % 2 != 0){
			if (i != 1){
				if (i <= 10){
					if (i == 2 || i == 3 || i == 5 || i == 7){
						printf("%d\n", i);
					}
				}
				else{
					w = (int)sqrt(i);
					y = sqrt(i);
					if (w != y){
						w++;
					}
					for (j = 2; j <= w; j++){
						if (i % j == 0){
							break;
						}
					}
					if (j == w+1){
						printf("%d\n", i);
					}
				}
			}
		}
	}
}
//에라스토테네스의 체x
